### PR TITLE
Plaid link metadata

### LINKED ISSUE #13: Link flow data structures are not included in the library
The Plaid Link flows contain data structures which do not exist in the current library.
These are incredibly useful to passing state and context from the front-end Javascript implementation to backend services or Blazor UX components.

### PR DESCRIPTION
**Pull request by thesuavehog**
Fixes #13 

Implemented data structures to cover the Plaid Link flows.
These are self-contained with no broad reach into other areas of the library.

The only slight quirk is that Plaid returns multiple items such as error & metadata, event name & metadata, etc. in some flows. It's much easier and more useful to just combine these in the Javascript side into 1 object so I have implemented these objects assuming that is done first.

The optional properties are nullable so they will all work if the user does not combine them first as well.

I have included documentation in the classes to illustrate the use of them via Javascript code.

### CODE REVIEW

**Comment by thesuavehog** [comment]
@viceroypenguin - I forgot about JsonStringEnumMemberConverter's possible null reference return. Any idea how to address that? ... or was this the conflict w System.Text.Json we'd previously discussed and never really had a clean solution for this issue?

**Comment by viceroypenguin** [comment]
@thesuavehog Did you see the email I sent you?

File: src/Plaid/Management/PlaidErrorCode.cs [hunk 1]
```diff
@@ -0,0 +1,306 @@
+﻿using System.ComponentModel;
```
Comment (viceroypenguin) [inline src/Plaid/Management/PlaidErrorCode.cs:1]: File should be removed in favor of existing `ErrorCode` enum.

File: src/Plaid/Management/PlaidErrorType.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
+﻿using System;
```
Comment (viceroypenguin) [inline src/Plaid/Management/PlaidErrorType.cs:1]: File should be removed in favor of existing `ErrorType` enum

File: src/Plaid/Plaid.csproj [hunk 3]
```diff
@@ -43,6 +43,7 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
+    <PackageReference Include="System.Text.Json" Version="4.7.2" />
```
Comment (viceroypenguin) [inline src/Plaid/Plaid.csproj:46]: Should remove this reference entirely unless necessary?

File: src/Plaid/Link/OnSuccessMetadata.cs [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+		/// <value>The link session identifier.</value>
+		[JsonProperty("account_id")]
+		public string AccountId { get; set; } = null!;
+
+		/// <summary>
+		/// Gets or sets the accounts info.
+		/// </summary>
+		/// <value>The accounts.</value>
+		[JsonProperty("accounts")]
+		[System.Diagnostics.CodeAnalysis.SuppressMessage(
```
Comment (viceroypenguin) [inline src/Plaid/Link/OnSuccessMetadata.cs:56]: Remove attribute, and switch to `IReadOnlyList<AccountInfo>`. We'll do another PR shortly to migrate other properties to `IReadOnlyList<>` as well.

File: src/Plaid/Link/OnSuccessMetadata.cs [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+
+		/// <summary>
+		/// Represents an <see cref="Entity.Account"/> metadata.
+		/// </summary>
+		[System.Diagnostics.CodeAnalysis.SuppressMessage(
+			"Design",
+			"CA1034:Nested types should not be visible",
+			Justification = "Required for Json deserialization?")]
+		public class AccountInfo
```
Comment (viceroypenguin) [inline src/Plaid/Link/OnSuccessMetadata.cs:90]: This appears to be a duplicate of the existing `Account` class. Any reason for this specifically?

File: src/Plaid/Link/OnSuccessMetadata.cs [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+
+		/// <summary>
+		/// Represents an <see cref="Entity.Institution"/> metadata.
+		/// </summary>
+		[System.Diagnostics.CodeAnalysis.SuppressMessage(
+			"Design",
+			"CA1034:Nested types should not be visible",
+			Justification = "Required for Json deserialization?")]
+		public class InstitutionInfo
```
Comment (viceroypenguin) [inline src/Plaid/Link/OnSuccessMetadata.cs:138]: This appears to be a duplicate of the existing `Institution` class. Any reason for this specifically?

File: src/Plaid/Link/OnExitMetadata.cs [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+		/// Gets or sets the institution.
+		/// </summary>
+		/// <value>The institution.</value>
+		[JsonProperty("institution")]
+		public InstitutionInfo Institution { get; set; } = null!;
+
+		/// <summary>
+		/// Represents an <see cref="Entity.Institution"/> metadata.
+		/// </summary>
+		public class InstitutionInfo
```
Comment (viceroypenguin) [inline src/Plaid/Link/OnExitMetadata.cs:84]: This appears to be a duplicate of the existing `Institution` class. Any reason for this specifically?

File: src/Plaid/Link/OnEventMetadata.cs [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+		///	  // }
+		///	  console.info('onEvent:' + eventName, metadata);
+		///	  metadata.event_name = eventName;
+		///	  ... send the metadata as JSON to be deserialized into this class ...  
+		///  }
+		/// </code>
+		/// </example>
+		/// </summary>
+		[JsonProperty("event_name")]
+		[Newtonsoft.Json.JsonConverter(typeof(Newtonsoft.Json.Converters.StringEnumConverter))]
```
Comment (viceroypenguin) [inline src/Plaid/Link/OnEventMetadata.cs:61]: Can we simplify names here? `Newtonsoft.Json` doesn't appear to be necessary, since we have the `using` statement for it. Add one for `Newtonsoft.Json.Converters` as well?

**Review by viceroypenguin — CHANGES_REQUESTED** [review]

File: src/Plaid/Link/OnEventMetadata.cs (hunk 8 above)
Comment (thesuavehog) [reply src/Plaid/Link/OnEventMetadata.cs:61]: oops. my bad

File: src/Plaid/Management/PlaidErrorCode.cs (hunk 1 above)
Comment (thesuavehog) [reply src/Plaid/Management/PlaidErrorCode.cs:1]: forgot I had those old ones still in there... will remove

File: src/Plaid/Management/PlaidErrorType.cs (hunk 2 above)
Comment (thesuavehog) [reply src/Plaid/Management/PlaidErrorType.cs:1]: forgot I had those old ones still in there... will remove

File: src/Plaid/Link/OnSuccessMetadata.cs (hunk 6 above)
Comment (thesuavehog) [reply src/Plaid/Link/OnSuccessMetadata.cs:138]: It's a subset of the properties plus an Id property. I know... friggin' annoying they didn't just use the same data structure

File: src/Plaid/Link/OnExitMetadata.cs (hunk 7 above)
Comment (thesuavehog) [reply src/Plaid/Link/OnExitMetadata.cs:84]: In theory we could reuse Institution but this data structure in their API only ever has 2 properties so I though it would be clearer when someone is trying to figure out why they are not getting the rest of the data, that it's only two properties.

File: src/Plaid/Link/OnSuccessMetadata.cs (hunk 6 above)
Comment (viceroypenguin) [reply src/Plaid/Link/OnSuccessMetadata.cs:138]: Derp! Ok, update comments in the xdocs, and add a remark about duplicate of the real type for such and such.

File: src/Plaid/Link/OnSuccessMetadata.cs [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+			public Going.Plaid.Entity.AccountSubType? SubType { get; set; } = null!;
+		}
+
+
+		/// <summary>
+		/// Represents an <see cref="Entity.Institution"/> metadata.
+		/// </summary>
+		[System.Diagnostics.CodeAnalysis.SuppressMessage(
+			"Design",
+			"CA1034:Nested types should not be visible",
```
Comment (viceroypenguin) [inline src/Plaid/Link/OnSuccessMetadata.cs:136]: if you update .editorconfig to have `dotnet_diagnostic.CA1034.severity = none` (matching the others in that file), can we remove this attribute?

File: src/Plaid/Link/OnSuccessMetadata.cs (hunk 6 above)
Comment (thesuavehog) [reply src/Plaid/Link/OnSuccessMetadata.cs:123]: oops.. I meant o put this comment on the AccountInfo feedback but, same comment anyway =)

File: src/Plaid/Link/OnSuccessMetadata.cs (hunk 5 above)
Comment (thesuavehog) [reply src/Plaid/Link/OnSuccessMetadata.cs:75]: It's a subset of the properties plus an Id property. I know... friggin' annoying they didn't just use the same data structure

**Review by viceroypenguin — APPROVED** [review]
